packages/zoltan2/src/directory/Zoltan2_Directory_Impl.hpp

```diff
@@ -50,16 +50,52 @@
 #include "Zoltan2_Directory.hpp"
 #include "Zoltan2_Directory_Comm.hpp"
 
-// Supplies the current hash - rolled over from zoltan
-// Right now the hash_proc method is still being used for the new Kokkos mode
-// which determines which proc will own a gid. TODO: Make this use a Kokkos
-// setup. The Kokkos mode is using the Kokkos unordered map for the node data.
-// I added the declaration here as inline to avoid the warnings generated
-// otherwise. All this should eventually go away and either use Kokkos methods
-// or implement the hash here so we don't have the links to zoltan
-static inline void MurmurHash3_x86_32 (
-  const void * key, int len, uint32_t seed, void * out);
-#include "murmur3.c"
+// Copied from murmurc.3
+// TODO: Will be removed as Kokkos form develops
+#define ZOLTAN2_ROTL32(x,r) (uint32_t) \
+  (((uint32_t)(x) << (int8_t)(r)) | ((uint32_t)(x) >> (32 - (int8_t)(r))))
+static inline void Zoltan2_MurmurHash3_x86_32 ( const void * key, int len,
```

Comment:
The idiomatic way to have a `static` (internal linkage) function in C++ is to put it in an anonymous namespace, like this:
```c++
namespace { // (anonymous)
  void foo () {
    whatever_code ();
  }
} // namespace (anonymous)
```

Reply:
I will fix this.